### PR TITLE
Add unattended install script.

### PR DESCRIPTION
**Pull request by ganoze**
# Pull Request Template

## Description

This PR adds a new script that aims at automating all manual steps that are required when installing Centreon from packages.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [X] 19.10.x (master)

Note : installation is currently made with the 19.04 version of Centreon as it is the latest stable release available now.

<h2> How this pull request can be tested ? </h2>

If the environment is supported, the script should execute without error and left you with a working Centreon server (but the web install still have to be executed).

If the environment is not supported the script should fail with an explicit error message.

`curl -L https://github.com/centreon/centreon/raw/MON-4142-simple-install/unattended.sh | sh`

#### Community contributors & Centreon team

- [X] I followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [X] I have made sure that the **unit tests** related to the story are successful.
- [X] I have made sure that **unit tests cover 80%** of the code written for the story.
- [X] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: unattended.sh [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+else
+  print_step_end "OK, systemd not detected, skipping"
+fi
+
+#
+# PHP
+#
+
+print_step_begin "PHP configuration"
+echo "date.timezone = Europe/Paris" > /etc/opt/rh/rh-php71/php.d/10-centreon.ini
```
Comment (kduret) [inline unattended.sh]: should not be rh-php72 ?

Comment (ganoze) [reply unattended.sh]: This PR is currently targeting the 19.04 serie that comes with PHP 7.1.

Comment (lgcosta) [inline unattended.sh]: Get the timezone from system, like as:
```
TIMEZONE=${$(readlink /etc/localtime)#*zoneinfo/}
```

File: unattended.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+print_step_end
+
+#
+# MARIADB
+#
+
+print_step_begin "MariaDB configuration"
+if [ "x$has_systemd" '=' x1 ] ; then
+  mkdir -p  /etc/systemd/system/mariadb.service.d
+  echo -e "[Service]\nLimitNOFILE=32000" >> /etc/systemd/system/mariadb.service.d/limits.conf
```
Comment (lgcosta) [inline unattended.sh]: This is no longer set by the rpm package (from Centreon repository)?

File: unattended.sh (hunk 1 above)
Comment (lgcosta) [reply unattended.sh]: More consistent:
```
[root@localhost ~]# timedatectl | grep "Time zone:" | awk '{ print $3 }'
Europe/Paris
```

Comment (ganoze) [reply unattended.sh]: Good improvement indeed. Fixed using `date '+%Z'` that is POSIX-compliant.

File: unattended.sh (hunk 2 above)
Comment (ganoze) [reply unattended.sh]: Indeed it is set by the _centreon-database_ package. I removed the whole _MariaDB_ section.

File: unattended.sh [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+fi
+
+#
+# REPOSITORY
+#
+
+print_step_begin "Centreon official repositories installation"
+rpm -q centos-release-scl > /dev/null 2>&1
+yum -q clean all
+if [ "x$?" '!=' x0 ] ; then
```
Comment (adr-mo) [inline unattended.sh]: This condition is written to make sure that centos-scl-release is installed. If not then install.
However you are using $? which corresponds to the returned code of the previous executed command ie `yum -q clean all`

I guess that the line 77 is meant to test the returned code of the command `rpm -q centos-release-scl`

File: unattended.sh [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+print_step_end
+
+#
+# CENTREON
+#
+
+print_step_begin "Centreon installation"
+yum -q install -y centreon
+if [ "x$?" '!=' x0 ] ; then
+  error_and_exit "Coult nod install Centreon (package centreon)"
```
Comment (adr-mo) [inline unattended.sh]: Small typo `"Could not"`

Comment (ganoze) [reply unattended.sh]: Fixed. thanks for the catch !

File: unattended.sh (hunk 3 above)
Comment (ganoze) [reply unattended.sh]: 100% truth. I fixed it by switching the lines.

Comment (ganoze) [reply unattended.sh]: Fixed by switching lines.

File: unattended.sh [hunk 5]
```diff
@@ -110,19 +110,34 @@ if [ -z "$timezone" ] ; then
...
 print_step_end "OK, timezone set to $timezone"
 
+#
+# FIREWALL
+#
+
+print_step_begin "Firewall configuration"
+command -v firewall-cmd > /dev/null 2>&1
+if [ "x$?" '=' x0 ] ; then
+  for svc in http mysql snmp snmptrap ; do
```
Comment (lgcosta) [inline unattended.sh]: I would not recommend leaving mysql port service open

Comment (ganoze) [reply unattended.sh]: Any opinion on this @lpinsivy ?

**Comment by ganoze** [comment]
@lgcosta IMHO triggering a full system upgrade is far beyond the goal of quickly installing Centreon.

During the RPM installation, the dependencies will update some system components so that Centreon works correctly. Updating other components besides that, is left to the system administrator.

However I implemented the Red Hat release checking mechanism. Thanks for the idea.

File: unattended.sh (hunk 5 above)
Comment (ganoze) [reply unattended.sh]: OK I removed _mysql_ from the service list.

**Pull request merged**